libc/watcom/asm/fchop87.asm

```diff
@@ -44,7 +44,7 @@
 include mdef.inc
 include math87.inc
 
-        xrefp           __8087  ; indicate that NDP instructions are present
+;        xrefp           __8087  ; indicate that NDP instructions are present
```

Comment:
This is fine for now, as I have not yet figured out how to use OWC floating point with ELKS without requiring a full compile/replacement of the OWC soft floating point and math libraries - which look like a very large effort. (If both the 8086 ASM routines and the OWC math library could be moved easily to libc/watcom/math and built, that would be great).

Currently, OWC applications running on ELKS require a hardware 8087+ in order to run, something that definitely won't work for some of our user base.  I was initially thinking of using `__8087` as a method of knowing whether floating point was used by the program, as my intention was to use weak symbol linking and have the stdio/printf library pull in `dtostr` only when floating point was used, to keep things small. This is work in progress. It is important there be a way to use weak linking when floating point is used in an application. Weak linking is supported by ia16-elf-gcc and ELKS [here](https://github.com/ghaerr/elks/blob/master/libc/include/sys/linksym.h), and I recently added support for weak linking with ELKS to OWC in the same file.

I tried setting up OWC with soft float but noticed that since it's not IEEE 754 compatible, the ELKS math library (libc/math) I ported from elsewhere won't work. Thus I have had to stick with hw float. I see that OWC has very fancy "8087-detection" support where apparently FPU instruction trap and some OS or library code replaces the FPU instruction with a software trap to a soft fp emulator. I deemed this as way too complicated for ELKS (at least at the moment). 

ia16-elf-gcc has automatic soft floating point, which is currently the default and probably what we'd prefer, even in its not IEEE 754 compliant. The demo program for this is [basic](https://github.com/ghaerr/elks/tree/master/elkscmd/basic). It now compiles and runs using hw float but no math library (see elkscmd/basic/host.h).

What are your thoughts on all this?

Reply:
I am not expert on the math and FPU that it is only my opinion. If I understand correctly then now ELKS has no support for FPU. Also support from compiler is needed. I don't know what is possible with ia16-gcc if it is able recognize use of  floating point and generate some symbols to signal linker to load some floating point stuff.
At minimum you need some hardware support for FPU in ELKS to handle FPU exceptions. Too many unclear things for me and also my knowledge in this area is limited.

Reply:
The ELKS kernel doesn't have any FPU support, it catches but ignores any FPU-generated exceptions. The kernel itself uses no floating point.

Application support for FP is soft-only using ia16-elf-gcc, and automatically provided, which is nice. I added a math library which works with the IEEE 754 used by the gcc basic soft float library.

What I would like to see is options for using OWC to compile applications with self-contained soft FP support, but I have not (yet) got that to work, as large OWC ASM library of routines need to be built as part of ELKS libc. Once this is done, likely OWC math library can also be built and problem solved. During this process, it would be nice if the kernel could tell whether an executable used soft vs hw float.

At this point, it is probably not important that ELKS use exception-based hard/soft FP support, although it might be nice to have a separate set of `ewcc` options that allow for compiling an application to require FPU vs soft-only.

Reply:
Approved. We will worry about soft vs hard float support in a separate conversation, thank you.